modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php

```diff
@@ -0,0 +1,873 @@
+
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\StringTextareaWidget;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\tide_content_collection\SearchApiIndexHelper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Implementation of the content collection configuration widget.
+ *
+ * @FieldWidget(
+ *   id = "content_collection_configuration",
+ *   label = @Translation("Content Collection Configuration"),
+ *   field_types = {
+ *     "content_collection_configuration"
+ *   }
+ * )
+ */
+class ContentCollectionConfigurationWidget extends StringTextareaWidget implements ContainerFactoryPluginInterface {
+
+  /**
+   * The Search API Index helper.
+   *
+   * @var \Drupal\tide_content_collection\SearchApiIndexHelper
+   */
+  protected $indexHelper;
+
+  /**
+   * The module handler service.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * The search API index.
+   *
+   * @var \Drupal\search_api\IndexInterface
+   */
+  protected $index;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, array $third_party_settings, ModuleHandlerInterface $module_handler, SearchApiIndexHelper $index_helper) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $third_party_settings);
+    $this->moduleHandler = $module_handler;
+    $this->indexHelper = $index_helper;
+    $this->getIndex();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['third_party_settings'],
+      $container->get('module_handler'),
+      $container->get('tide_content_collection.search_api.index_helper')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+      'tabs' => [
+        'content' => [
+          'content_types_wrapper' => [
+            'enable_content_types' => '1',
+            'contentTypes' => [],
+          ],
+          'field_topic_wrapper' => [
+            'enable_field_topic' => '1',
+            'show_field_topic_operator' => 0,
+          ],
+          'field_tags_wrapper' => [
+            'enable_field_tags' => '1',
+            'show_field_tags_operator' => 0,
+          ],
+        ],
+        'enable_call_to_action' => 0,
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() : array {
+    $summary = [];
```

Comment:
No need to output everything in the widget settings summary.

Reply:
Cleared the summary. Nothing shown.